web/pdf_link_service.js

```diff
@@ -549,59 +546,53 @@ class PDFLinkService {
     return this.#pagesRefCache.get(refStr) || null;
   }
 
-  static #isValidExplicitDestination(dest) {
-    if (!Array.isArray(dest)) {
-      return false;
-    }
-    const destLength = dest.length;
-    if (destLength < 2) {
+  static #isValidExplicitDest(dest) {
+    if (!Array.isArray(dest) || dest.length < 2) {
       return false;
     }
-    const page = dest[0];
+    const [page, zoom, ...args] = dest;
     if (
       !(
         typeof page === "object" &&
-        Number.isInteger(page.num) &&
-        Number.isInteger(page.gen)
+        Number.isInteger(page?.num) &&
+        Number.isInteger(page?.gen)
       ) &&
-      !(Number.isInteger(page) && page >= 0)
```

Comment:
`Number.isInteger(-1)` is also true, so shouldn't we still guard against negative page numbers here (and also in the mostly-copied worker variant above)?

Reply:
We have an old guard, which should be sufficient to prevent any issues here: https://github.com/mozilla/pdf.js/blob/7d1eabe2bd8e8becc37a1d0051bb2e9b361104cf/web/pdf_link_service.js#L214-L220

Moving the `page >= 0` check into the worker caused some annotation unit-tests to fail, and given the overall changes in this patch I'd rather err on the side of not imposing *too many* restrictions here (for a "harmless" case).

Reply:
I agree; this should then be sufficient, and otherwise we can always amend this later if we actually run into such cases (which I think to this date we haven't actually seen).